Tests/NIOExtrasTests/LengthFieldBasedFrameDecoderTest.swift

```diff
@@ -455,4 +455,31 @@ class LengthFieldBasedFrameDecoderTest: XCTestCase {
             })
         }
     }
+    
+    func testMaliciousLengthWith32BitFieldLength() throws {
+        self.decoderUnderTest = .init(LengthFieldBasedFrameDecoder(lengthFieldLength: .four,
+                                                                   lengthFieldEndianness: .little))
+        XCTAssertNoThrow(try self.channel.pipeline.addHandler(self.decoderUnderTest).wait())
+
+        let dataLength = UInt32(Int32.max) + 1
+        
+        var buffer = self.channel.allocator.buffer(capacity: 4) // 4 byte header
+        buffer.writeInteger(dataLength, endianness: .little, as: UInt32.self)
+        buffer.writeString(standardDataString)
+        
+        XCTAssertThrowsError(try self.channel.writeInbound(buffer))
+    }
+    
+    func testMaliciousLengthWith64BitFieldLength() {
+        self.decoderUnderTest = .init(LengthFieldBasedFrameDecoder(lengthFieldLength: .eight,
+                                                                   lengthFieldEndianness: .little))
+        XCTAssertNoThrow(try self.channel.pipeline.addHandler(self.decoderUnderTest).wait())
+
+        let dataLength = UInt64(Int64.max) + 1
+        
+        var buffer = self.channel.allocator.buffer(capacity: 8) // 8 byte header
+        buffer.writeInteger(dataLength, endianness: .little, as: UInt64.self)
+        
+        XCTAssertThrowsError(try self.channel.writeInbound(buffer))
+    }
```

Comment:
if you don't mind, we could add maybe two more edge-cases here which make sure we can correctly parse the maximum allowable value (`Int32.max`) and we fail with the minimum not-allowable (`Int32.max + 1`). For the second point, we can probably modify the tests above.

Reply:
Sure :) added